src/test/java/io/neonbee/entity/UnregisterEntitiesTest.java

```diff
@@ -0,0 +1,83 @@
+import static io.neonbee.NeonBeeInstanceConfiguration.ClusterManager.INFINISPAN;
+import static io.neonbee.NeonBeeProfile.WEB;
+import static io.vertx.core.Future.succeededFuture;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.olingo.commons.api.edm.FullQualifiedName;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.neonbee.NeonBee;
+import io.neonbee.NeonBeeExtension;
+import io.neonbee.NeonBeeInstanceConfiguration;
+import io.neonbee.internal.cluster.ClusterHelper;
+import io.neonbee.test.helper.DeploymentHelper;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxTestContext;
+
+@ExtendWith({ NeonBeeExtension.class })
+class UnregisterEntitiesTest {
+
+    @Test
+    @Timeout(value = 10, timeUnit = TimeUnit.SECONDS)
+    @DisplayName("test unregistering entity models (Infinispan cluster)) ")
+    void testInfinispanUnregisteringEntities(@NeonBeeInstanceConfiguration(activeProfiles = WEB, clustered = true,
+            clusterManager = INFINISPAN) NeonBee web, VertxTestContext testContext) {
+        testUnregisteringEntities(web, testContext);
+    }
+
+    @Test
+    @Timeout(value = 20, timeUnit = TimeUnit.SECONDS)
+    @DisplayName("test unregistering entity models (Hazelcast cluster)")
+    void testHazelcastUnregisteringEntities(@NeonBeeInstanceConfiguration(activeProfiles = WEB, clustered = true,
+            clusterManager = HAZELCAST) NeonBee web, VertxTestContext testContext) {
+        testUnregisteringEntities(web, testContext);
+    }
+
+    private void testUnregisteringEntities(NeonBee web, VertxTestContext testContext) {
+        assertThat(isClustered(web)).isTrue();
+
+        Vertx vertx = web.getVertx();
+        String hazelcastNodeId = ClusterHelper.getNodeId(web.getVertx());
+
+        vertx.deployVerticle(new EntityVerticleUnregisterImpl()).onSuccess(unused -> testContext.verify(() -> {
+            assertThat(DeploymentHelper.isVerticleDeployed(vertx, EntityVerticleUnregisterImpl.class)).isTrue();
+        })).compose(unused -> EntityVerticle.getNodeEntitiesMap(vertx)
+                .compose(map -> map.get(ClusterHelper.getNodeId(web.getVertx())))
+                .onSuccess(ja -> testContext.verify(() -> {
+                    JsonArray jsonArray = (JsonArray) ja;
+                    assertThat(jsonArray).hasSize(2);
+                }))).compose(unused -> UnregisterEntityModelHook.unregister(vertx, hazelcastNodeId))
+                .compose(unused -> web.getAsyncMap()
+                        .get(EntityVerticle.sharedEntityMapName(EntityVerticleUnregisterImpl.FQN_ERP_CUSTOMERS)))
+                .onSuccess(jsonArray -> testContext.verify(() -> {
+                    assertThat(jsonArray).isEqualTo(new JsonArray());
```

Comment:
isEmpty()?

Reply:
I had to cast jsonArray to an iterable. But to be honest I think the generated error message is better this way.